### PR TITLE
allow to preserve attributes on extend[Foreign]Object

### LINKED ISSUE #1183: Preserve name field on instance object creation/update
When we update instance objects (or objects configured in io-package.json) we should preserve the common.name field ... maybe add that as a general option to extendObject and use extendObject for object updates/creations.

needed for clean solution for https://github.com/Apollon77/ioBroker.daikin/issues/54

### PR DESCRIPTION
**Pull request by foxriver76**
- and use this for `instancesObjects` creation at instance startage
- closes #1183

### CODE REVIEW

**Comment by Apollon77** [comment]
maybe we should check the 2 LGTM comments :-)

File: lib/adapter.js [hunk 1]
```diff
@@ -2040,6 +2046,28 @@ function Adapter(options) {
...
 
+            // remove the preserve attributes
+            if (oldObj && options && tools.isObject(options.preserve)) {
+                /**
+                 * Removes properties which are given by preserve
+                 * @param {object} preserve - object which has true entries for all attributes which should be removed from currObj
+                 * @param {object} oldObj - old object
+                 * @param {object} newObj - new object
+                 */
+                const rmProperties = (preserve, oldObj, newObj) => {
```
Comment (Apollon77) [inline lib/adapter.js:2057]: Why not put it one level above and use same for both places?

Comment (foxriver76) [reply lib/adapter.js:2057]: hm very specific method only interesting for this extendObject, but we can also provide it outside of the methods

Comment (Apollon77) [reply lib/adapter.js:2057]: I was justthinking because of the duplicated code
@AlCalzone what do you think?

Comment (Apollon77) [reply lib/adapter.js:2057]: We could also put it into tools.js :-)

Comment (AlCalzone) [reply lib/adapter.js:2057]: sounds like a decent utility method for tools.

File: lib/tools.js [hunk 2]
```diff
@@ -2379,6 +2379,23 @@ function parseShortGithubUrl(url) {
     };
 }
 
+/**
+ * Removes properties which are given by preserve
+ * @param {object} preserve - object which has true entries for all attributes which should be removed from currObj
+ * @param {object} oldObj - old object
+ * @param {object} newObj - new object
```
Comment (AlCalzone) [inline lib/tools.js:2386]: On a side note, `object` is equivalent to `any`

Comment (foxriver76) [reply lib/tools.js:2386]: afaik there is no real alternative without defining each attribute of the object separately

Comment (foxriver76) [reply lib/tools.js:2386]: besides ts ;-)

Comment (AlCalzone) [reply lib/tools.js:2386]: `Record<string, any>`

Comment (AlCalzone) [reply lib/tools.js:2386]: won't help much in this case though

Comment (foxriver76) [reply lib/tools.js:2386]: I don’t find a valid documentation of Record for plain jsdoc but would be interesting for future implementation s

**Pull request merged**